plan.go

```diff
@@ -40,17 +46,23 @@ func (p *plan) Replace(ctx context.Context, from interface{}, to interface{}) er
 	}
 
 	for i := range p.order {
-		b := p.order[i]
-		if f.Name == b.Name {
-			// same function, lets replace it
-			p.order[i] = t
-			return nil
+		for j := range p.order[i] {
+			b := p.order[i][j]
+			if f.Name == b.Name {
+				// same function, lets replace it
+				p.order[i][j] = t
+				return nil
+			}
 		}
 	}
 	return errors.New("builder not found")
 }
 
 func (p *plan) Run(ctx context.Context, initData ...interface{}) (Result, error) {
+	return p.RunParallel(ctx, 1, initData...)
+}
+
+func (p *plan) RunParallel(ctx context.Context, workers uint, initData ...interface{}) (Result, error) {
```

Comment:
I'm wondering if a go routine leak is possible. I added `goleak.VerifyNone(t)` (using https://github.com/uber-go/goleak) to `TestRunPlan` and got this error:

```
    leaks.go:78: found unexpected goroutines:
        [Goroutine 30 in state chan receive, with github.com/go-coldbrew/data-builder.worker on top of the stack:
        goroutine 30 [chan receive]:
        github.com/go-coldbrew/data-builder.worker({0x47ac9d0, 0xc000016148}, 0xc0001b8680?)
        	/Users/peter/dev/data-builder/plan.go:103 +0x7d
        created by github.com/go-coldbrew/data-builder.(*plan).run
        	/Users/peter/dev/data-builder/plan.go:176 +0x9a
        ]
```

Reply:
let me fix it